cmdb/framework/managers/object_manager.py

```diff
@@ -109,3 +136,41 @@ def iterate(self, filter: dict, limit: int, skip: int, sort: str, order: int,
         iteration_result: IterationResult[CmdbObject] = IterationResult.from_aggregation(aggregation_result)
         iteration_result.convert_to(CmdbObject)
         return iteration_result
+
+    def references(self, object_: CmdbObject, filter: dict, limit: int, skip: int, sort: str, order: int,
+                   user: UserModel = None, permission: AccessControlPermission = None, *args, **kwargs) \
+            -> IterationResult[CmdbObject]:
+        query = []
+        if isinstance(filter, dict):
+            query.append(filter)
+        elif isinstance(filter, list):
+            query += filter
+        query.append({
+            '$lookup': {
+                'from': 'framework.types',
+                'localField': 'type_id',
+                'foreignField': 'public_id',
+                'as': 'type'
+            }
+        })
+        query.append({
+            '$unwind': {
+                'path': '$type'
+            }
+        })
+        query.append({
+            '$match': {
+                'type.fields.type': 'ref',
+                '$or': [
+                    {'type.fields.ref_types': Regex(f'.*{object_.type_id}.*', 'i')},
```

Comment:
If I understand correctly you want to start here all object IDs that can be in an array. Can it be that it is error-prone? Are values like -> (1, 3, 5) or ['1', 3, '5'] accepted if the searched ID is 5?

Would prefer something like this:
{ "$expr": { "$in": ['object_.type_id', ''type.fields.ref_types''] }}

Reply:
As discussed, we'll take it from there for now, and look at it when we switch over from the object REST API.